### ISSUE TITLE
Last draw with poly command not completed

### ISSUE THREAD

**Issue by karlmsmith** [NOAA-PMEL/Ferret#1477]
**Reported by karl smith on 7 Oct 2014 22:13 UTC**
With the poly command, the last drawing piece is not completed.  The plot command does NOT have this problem.
So
```
set win /asp=1 1
let v = 2*3.14159*x[gx=0:1:.01]
poly /hl=-1:1 /vl=-1:1 /pal=red sin(v), cos(v)
```
is missing the cross of the final t in the title "constant".  Adding ``` /title="constant." ``` adds the cross to the t but leaves off the period.  Instead adding ``` /title ``` (so no title) leaves off the closing parenthesis on the X-axis label "SIN(V)".  Add a call to GUWK (update workstation) at the end of PLTIT makes no difference, so it appears the last drawing command is not actually being called.

If ``` plot /vs ``` is used instead of ``` poly ```, everything works fine; no missing pieces.

Came out of ticket #2086 reported by Andrew.

Migrated-From: http://dunkel.pmel.noaa.gov/trac/ferret/ticket/2205

**Comment by karlmsmith** [comment]
**Comment by ansley.b.manke on 7 Oct 2014 22:32 UTC**
I wonder if this could be the cause of some of our other POLYGON bugs. #1751, #986 seem like possibilities.

**Comment by karlmsmith** [comment]
**Modified by andrew.wittenberg on 8 Oct 2014 19:14 UTC**

**Comment by karlmsmith** [comment]
**Comment by ansley.b.manke on 9 Oct 2014 18:34 UTC**
This is not in the polygon routines but in the setup portion of the plot where axes and axis labels are drawn. Polygon plots have some special setup because the polygons themselves may overdraw plot axes. See #1663 and the other tickets it references.

I've added a call to redraw the main plot label for polygon plots.

**Comment by karlmsmith** [comment]
**Comment by karl.smith on 9 Oct 2014 21:25 UTC**
When a title is given, looks good.  But when no title is given (adding ```/title``` to the polygon command), the closing parenthesis on the X-axis label is still missing.  Maybe different priority on this; not sure how often folks do not want Ferret to give a title.

**Comment by karlmsmith** [comment]
**Comment by ansley.b.manke on 9 Oct 2014 22:02 UTC**
I've found a better fix, which takes care of both of these cases. I also tested that the axis labels are still complete if we specify /NOLABELS so that there are only the numeric axis-tic labels.

**Comment by karlmsmith** [comment]
**Comment by andrew.wittenberg on 9 Oct 2014 22:10 UTC**
Good catch, Karl, and thanks Ansley for your latest fix.

Does this fix the underlying problem, and somehow flush out that last drawing command?  We do frequently suppress the automatic titles and labels, so that we can apply one or more with specific positions & sizes using the LABEL command.  We've even suppressed the axes altogether, so that the only thing drawn is the data itelf.

Ansley, does your fix guarantee that the last element of the user's plot, whatever it may be, will always be drawn?

**Comment by karlmsmith** [comment]
**Comment by karl.smith on 9 Oct 2014 23:15 UTC**
A test script of only a few polygons:
```
set win /asp=0.67 1
let xvals = { 0.0, 0.866, 0.866, 0.0, -0.866, -0.866, 0.0,\
            , 0.866, 2.134, 2.134, 0.866, 0.866, \
            , 3.0, 3.866, 3.866, 3.0, 2.134, 2.134, 3.0, \
            , 0.0, 0.866, 2.134, 3.0, 0.0, \
            , 0.0, 0.866, 2.134, 3.0, 0.0 }
let yvals = { 1.0, 0.5, -0.5, -1.0, -0.5, 0.5, 1.0, \
            , 0.5, 0.5, -0.5, -0.5, 0.5, \
            , 1.0, 0.5, -0.5, -1.0, -0.5, 0.5, 1.0, \
            , 1.0, 0.5, 0.5, 1.0, 1.0, \
            , -1.0, -0.5, -0.5, -1.0, -1.0 }
let cvals = { 1.0, 2.0, 3.0, 4.0, 5.0 }
poly /noaxis /nolabel /nokey  xvals,yvals,cvals
```
does draw all the polygons.  Adding ``` /line /thick /fill ``` also draws all the outlines.  So this does appear to only be an issues with the labels.

**Comment by karlmsmith** [comment]
**Comment by ansley.b.manke on 9 Oct 2014 23:30 UTC**
My fix flushes the lines and text that's drawn for plot axes, axis labels, and the main plot label. All of these calls come before any of the polygon graphics calls are made.

**Comment by karlmsmith** [comment]
**Comment by andrew.wittenberg on 10 Oct 2014 00:36 UTC**
Great -- sounds like it's fixed then.

**Comment by karlmsmith** [comment]
**Comment by andrew.wittenberg on 30 Oct 2014 00:25 UTC**
Replying to [ansley.b.manke](comment:1):
> I wonder if this could be the cause of some of our other POLYGON bugs. #1751, #986 seem like possibilities.

It appears that the fix to this bug (#2205) has not resolved #1751 and #986.
